### PR TITLE
Reintroduce validate_seal method of Chain class

### PR DESCRIPTION
**Pull request by jannikluhn**
### What was wrong?

`MainnetTesterChain` was overriding `validate_seal` to do nothing, but this had no effect as it was not being used by `validate_block`.

### How was it fixed?

Add `validate_seal` in the `Chain` again.

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

**Comment by cburgdorf** [comment]
I *think* rather than reintroducing `validate_seal` on the `Chain` class, the `MainnetTesterChain` should rather ensure that `validate_seal` is monkey patched on the VMs.

**Comment by jannikluhn** [comment]
@cburgdorf I can see your point: `validate_seal` feels like an internal method that a subclass should not rely on. But monkey-patching the VMs also sounds kind of dirty. So I'd rather leave it like this, also considering that it originally was supposed to work like this.

File: evm/chains/base.py [hunk 1]
```diff
@@ -606,12 +610,17 @@ def validate_block(self, block: BaseBlock) -> None:
...
-
+        self.validate_seal(block.header)
         self.validate_uncles(block)
         self.validate_gaslimit(block.header)
 
+    def validate_seal(self, header: BlockHeader) -> None:
+        """
+        Validate the seal on the given header.
+        """
+        vm = self.get_vm()
```
Comment (carver) [inline evm/chains/base.py]: I believe this should be `self.get_vm(header)` so that we validate according to the correct fork's rules.

Comment (carver) [reply evm/chains/base.py]: Actually, I'm already working in the area, so I'll take care of it.

**Pull request merged**